### ISSUE TITLE
class hierarchy no longer correct

### ISSUE THREAD

**Issue by KrisThielemans** [jaeandersson/swig#95]
I've been using my [own fork](https://github.com/KrisThielemans/swig) but want to switch over to the most recent version. Unfortunately, the generated .m files no longer contain the correct class hierarchy for me, i.e. all classes are just derived from `SwigRef`, while this was working with my fork.

Our full project is at https://github.com/UCL/STIR but it's a bit too complicated as a test case. I'll try to make a simpler one. However, I first wanted to check if it works for others, and if I'm doing something stupid.

This happens even with non-templated classes.

The Python side is still working of course.

**Comment by KrisThielemans** [comment]
The test case was a bit on the trivial side. Here's my .i file:
```swig
%module testit
%{
#define SWIG_FILE_WITH_INIT
%}

%init %{
%}

%inline %{
struct A
{  int a;
};
struct B: public A
{  int b;
};
%}
```
command line
```sh
swig -matlab -c++  -module testit testit.i
```
First line of the generated B.m:
```
classdef B < SwigRef
```
(should be derived from `testit.A` of course)

**Comment by KrisThielemans** [comment]
will be fixed after #96
